firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHTTPClient.java

```diff
@@ -0,0 +1,233 @@
+package com.google.firebase.remoteconfig.internal;
+
+import static com.google.firebase.remoteconfig.FirebaseRemoteConfig.TAG;
+
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.android.gms.common.util.AndroidUtilsLight;
+import com.google.android.gms.common.util.Hex;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.installations.FirebaseInstallationsApi;
+import com.google.firebase.installations.InstallationTokenResult;
+import java.util.Collections;
+import java.util.EventListener;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import okhttp3.ConnectionPool;
+import okhttp3.OkHttpClient;
+import okhttp3.Protocol;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.sse.EventSource;
+import okhttp3.sse.EventSourceListener;
+import okhttp3.sse.EventSources;
+
+public class ConfigRealtimeHTTPClient {
+    // Headers
+    private static final String API_KEY_HEADER = "X-Goog-Api-Key";
+    private static final String X_ANDROID_PACKAGE_HEADER = "X-Android-Package";
+    private static final String X_ANDROID_CERT_HEADER = "X-Android-Cert";
+    private static final String X_GOOGLE_GFE_CAN_RETRY = "X-Google-GFE-Can-Retry";
+    private static final String INSTALLATIONS_AUTH_TOKEN_HEADER =
+            "X-Goog-Firebase-Installations-Auth";
+    
+    private static final String REALTIME_URL_STRING = "http://10.0.2.2:5000/sse";
+    private static final Logger logger = Logger.getLogger("Real_Time_RC");
+
+    private final ConfigFetchHandler configFetchHandler;
+    private final FirebaseApp firebaseApp;
+    private final FirebaseInstallationsApi firebaseInstallations;
+    private final Executor executor;
+    private final Context context;
+
+    // Retry parameters
+    private final Timer timer;
+    private final int ORIGINAL_RETRIES = 7;
+    private final long RETRY_TIME = 10000;
```

Comment:
You should include the unit that you are using in the name.

Reply:
Added seconds to param.